### PR TITLE
fix: ensure expressions that expect bools return bools

### PR DESCRIPTION
**Pull request by aliscott**
If a logical expression (|| or &&) had a sub-expression that was mocked to a non-boolean value, this would always mean the expression always returned false. We now handle this in the same way that we had condition expressions to ensure the mocked expressions return a boolean value if the operator expects boolean values.

### CODE REVIEW

File: internal/hcl/attribute.go [hunk 1]
```diff
@@ -594,16 +594,51 @@ func mockExpressionCalls(expr hcl.Expression, diagnostics hcl.Diagnostics, mocke
 			MarkerRange: t.MarkerRange,
 		}
 	case *hclsyntax.BinaryOpExpr:
+		// Logical operators (|| and &&) expect bools on both sides, so we need to mock the expression
+		// with a bool value.
+		impl := t.Op.Impl
+		params := impl.Params()
+		lhsParam := params[0]
+		rhsParam := params[1]
```
Comment (hugorut) [inline internal/hcl/attribute.go:602]: is this always expected to be a slice of this size?

Comment (hugorut) [reply internal/hcl/attribute.go:602]: we should make add a size check just in case?

**Review by hugorut — COMMENTED** [review]
looks good - minor question about the params size

File: internal/hcl/attribute.go (hunk 1 above)
Comment (aliscott) [reply internal/hcl/attribute.go]: Yeah, all BinaryOp functions have 2 params, but updated this anyway to make sure.

**Pull request merged**